flexmeasures/data/models/reporting/tests/test_pandas_reporter.py

```diff
@@ -159,3 +159,54 @@ def test_reporter_empty(setup_dummy_data):
     )
 
     assert report[0]["data"].empty
+
+
+def test_pandas_reporter_unit_conversion(app, setup_dummy_data):
+    """
+    Check that the unit conversion feature can handle the following cases:
+        - kW -> kW
+        - kW -> MW
+        - kW -> MWh
+    """
+    s1, s2, s3, s4, report_sensor, daily_report_sensor = setup_dummy_data
+
+    reporter_config = dict(
+        required_input=[
+            {"name": "sensor_4_kw"},
+            {"name": "sensor_4_mw", "unit": "MW"},
+            {"name": "sensor_4_mwh", "unit": "MWh"},
+        ],
+        required_output=[
+            {"name": "sensor_4_kw"},
+            {"name": "sensor_4_mw"},
+            {"name": "sensor_4_mwh"},
+        ],
+        transformations=[],
+    )
+
+    reporter = PandasReporter(config=reporter_config)
+
+    start = datetime(2023, 1, 1, tzinfo=utc)
+    end = datetime(2023, 1, 2, tzinfo=utc)
+    input = [
+        dict(name="sensor_4_kw", sensor=s4),
+        dict(name="sensor_4_mw", sensor=s4),
+        dict(name="sensor_4_mwh", sensor=s4),
+    ]
+    output = [
+        dict(name="sensor_4_kw", sensor=s4),
+        dict(name="sensor_4_mw", sensor=s4),
+        dict(name="sensor_4_mwh", sensor=s4),
```

Comment:
This also saves the results under sensor 4, right? Do we also convert the results back to its unit?

Reply:
The saving happens on the CLI command and, eventually, will happen on the job. The output of the reporter contains the sensor to which save the data to.

If we wanted to also convert the results of an output to the units of the sensor, we should tag the units of the transformations, at least, the last one. Alternatively, we could define the unit of the results in the  `required_output`. What do you think?

Reply:
Using the `required_output` makes sense, I think. It's then the job of whoever writes the reporter config to get the units right, and whoever writes the input/output just needs to pass a sensor to make the reporter convert to the sensor's units.